src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/OcrValidator.java

```diff
@@ -103,10 +103,12 @@ private void handleValidationResponse(
         log.info("Validation result for envelope {}: {}", envelope.zipFileName, res.status);
         switch (res.status) {
             case ERRORS:
+                String message = "OCR validation service returned OCR-specific errors. "
+                    + "Document control number: " + docWithOcr.documentControlNumber + ". "
+                    + "Envelope: " + envelope.zipFileName + ".";
                 throw new OcrValidationException(
-                    "OCR validation service returned OCR-specific errors. "
-                        + "Document control number: " + docWithOcr.documentControlNumber + ". "
-                        + "Envelope: " + envelope.zipFileName + "."
+                    message,
+                    message + "Errors: " + res.errors
```

Comment:
Good idea with the fields that contains detailed information.

This message is only good as the official `message` in OcrValidationException. `detailMessage` should just contain errors, as the message that we have is only good for our logging. In particular, the `OCR validation service returned OCR-specific errors` bit will not be understood by our supplier.

Errors should also be properly formatted, i.e. we should explicitly build a string from them using String.join(...) or something similar.

Reply:
Or we could start `detailMessage` with something like: `Service {} returned following OCR field validation errors:`. Still without document control number and envelope, because those will be known.

Reply:
`OCR validation service returned OCR-specific errors. Document control number: 852352234039153252048. Envelope: 59423325200994520_15-02-2019-08-08-19.zip. Errors: [first_name is missing, last_name is missing]`

this how it looks with our service

Reply:
> Or we could start `detailMessage` with something like: `Service {} returned following OCR field validation errors:`. Still without document control number and envelope, because those will be known.

does provider care about service?